wemake_python_styleguide/visitors/ast/loops.py

```diff
@@ -262,5 +262,19 @@ def _check_await_inside_loop(self, node: ast.Await) -> None:
         ):
             # async comprehensions are allowed to use `await`
             return
-        if node_parent is not None:
-            self.add_violation(AwaitInLoopViolation(node))
+
+        if node_parent is None:
+            return
+
+        if isinstance(node_parent, ast.For) and node_parent.iter is node:
+            # await allowed in loop definition
+            return
+
+        if isinstance(
```

Comment:
It would make sense to combine two `if isinstance(node_parent, AnyComprehension)` checks